File: people/kaiemjoy/cv/extract_publications.py
#!/usr/bin/env python3
"""
Extract publications for Dr. Kristen Aiemjoy from the main publications.yml file
"""

import yaml
import sys

def format_authors(authors):
    """Format author list with Aiemjoy in bold markers"""
    author_strs = []
    for author in authors:
        name = f"{author.get('given', '')} {author.get('family', '')}".strip()
        if author.get('family') == 'Aiemjoy':
            name = f"**{name}**"
        author_strs.append(name)
    return ', '.join(author_strs)

def extract_aiemjoy_publications(input_file='publications.yml', output_file='people/kaiemjoy/cv/publications.yml'):
    """
    Extract publications where Aiemjoy is an author
    """
    try:
        with open(input_file, 'r') as f:
            # Read the YAML file
            content = f.read()
            publications = yaml.safe_load(content)
        
        if not publications:
            print("No publications found in input file", file=sys.stderr)
            return
        
        # Filter publications where Aiemjoy is an author
        aiemjoy_pubs = []
        for pub in publications:
            if 'author' in pub:
                authors = pub['author']
                for author in authors:
                    if isinstance(author, dict) and author.get('family') == 'Aiemjoy':
                        # Add formatted author field
                        pub['authors_formatted'] = format_authors(pub['author'])
                        # Extract year from issued
                        if 'issued' in pub:
                            issued = pub['issued']
                            if isinstance(issued, str):
                                pub['year'] = issued[:4]
                            else:
                                pub['year'] = str(issued)[:4] if issued else ''
                        aiemjoy_pubs.append(pub)
                        break
        
        print(f"Found {len(aiemjoy_pubs)} publications for Dr. Aiemjoy", file=sys.stderr)
        
        # Write to output file
        with open(output_file, 'w') as f:
            yaml.dump(aiemjoy_pubs, f, default_flow_style=False, sort_keys=False, allow_unicode=True)
        
        print(f"Saved to {output_file}", file=sys.stderr)
        
    except FileNotFoundError:
        print(f"File {input_file} not found", file=sys.stderr)
        sys.exit(1)
    except yaml.YAMLError as e:
        print(f"Error parsing YAML: {e}", file=sys.stderr)
        sys.exit(1)
    except Exception as e:
        print(f"Unexpected error: {e}", file=sys.stderr)
        sys.exit(1)

if __name__ == "__main__":
    extract_aiemjoy_publications()

File: people/kaiemjoy/cv/extract_publications.R
#!/usr/bin/env Rscript
#
# Extract publications for Dr. Kristen Aiemjoy from the main publications.yml file
#

library(yaml)

#' Format author list with Aiemjoy in bold markers
#' 
#' @param authors List of author objects
#' @return Character string of formatted authors
format_authors <- function(authors) {
  author_strs <- sapply(authors, function(author) {
    given <- ifelse(is.null(author$given), "", author$given)
    family <- ifelse(is.null(author$family), "", author$family)
    name <- trimws(paste(given, family))
    
    if (!is.null(author$family) && author$family == "Aiemjoy") {
      name <- paste0("**", name, "**")
    }
    name
  })
  
  paste(author_strs, collapse = ", ")
}

#' Extract publications where Aiemjoy is an author
#' 
#' @param input_file Path to input publications.yml file
#' @param output_file Path to output publications.yml file
extract_aiemjoy_publications <- function(input_file = "publications.yml", 
                                         output_file = "people/kaiemjoy/cv/publications.yml") {
  tryCatch({
    # Read the YAML file
    publications <- yaml::read_yaml(input_file)
    
    if (is.null(publications) || length(publications) == 0) {
      message("No publications found in input file")
      return(invisible())
    }
    
    # Filter publications where Aiemjoy is an author
    aiemjoy_pubs <- list()
    
    for (pub in publications) {
      if (!is.null(pub$author)) {
        for (author in pub$author) {
          if (!is.null(author$family) && author$family == "Aiemjoy") {
            # Add formatted author field
            pub$authors_formatted <- format_authors(pub$author)
            
            # Extract year from issued
            if (!is.null(pub$issued)) {
              issued <- pub$issued
              tryCatch({
                if (is.character(issued)) {
                  pub$year <- substr(issued, 1, 4)
                } else if (is.numeric(issued)) {
                  pub$year <- as.character(issued)
                } else {
                  year_str <- as.character(issued)
                  pub$year <- ifelse(nchar(year_str) >= 4, substr(year_str, 1, 4), "")
                }
              }, error = function(e) {
                pub$year <- ""
              })
            }
            
            aiemjoy_pubs[[length(aiemjoy_pubs) + 1]] <- pub
            break
          }
        }
      }
    }
    
    message("Found ", length(aiemjoy_pubs), " publications for Dr. Aiemjoy")
    
    # Write to output file
    yaml::write_yaml(aiemjoy_pubs, output_file)
    
    message("Saved to ", output_file)
    
  }, error = function(e) {
    if (grepl("cannot open", e$message)) {
      message("File ", input_file, " not found")
      quit(status = 1)
    } else if (grepl("yaml", tolower(e$message))) {
      message("Error parsing YAML: ", e$message)
      quit(status = 1)
    } else {
      message("Unexpected error: ", e$message)
      quit(status = 1)
    }
  })
}

# Run main function if script is executed directly
if (!interactive()) {
  extract_aiemjoy_publications()
}
